src/mlpack/tests/rbm_network_test.cpp

```diff
@@ -0,0 +1,250 @@
+using namespace mlpack::optimization;
+using namespace mlpack::regression;
+
+BOOST_AUTO_TEST_SUITE(RbmNetworkTest);
+BOOST_AUTO_TEST_CASE(ClassificationTest)
+{
+  // Normalised dataset
+  int hiddenLayerSize = 100;
+  size_t batchSize = 10;
+  size_t numEpoches = 30;
+  arma::mat trainData, testData, dataset;
+  arma::mat trainLabelsTemp, testLabelsTemp;
+  trainData.load("digits_train.arm");
+  testData.load("digits_test.arm");
+  trainLabelsTemp.load("digits_train_label.arm");
+  testLabelsTemp.load("digits_test_label.arm");
+
+  arma::Row<size_t> trainLabels = arma::zeros<arma::Row<size_t>>(1,
+      trainLabelsTemp.n_cols);
+  arma::Row<size_t> testLabels = arma::zeros<arma::Row<size_t>>(1,
+      testLabelsTemp.n_cols);
+
+  for (size_t i = 0; i < trainLabelsTemp.n_cols; ++i)
+    trainLabels(i) = arma::as_scalar(trainLabelsTemp.col(i));
+
+  for (size_t i = 0; i < testLabelsTemp.n_cols; ++i)
+    testLabels(i) = arma::as_scalar(testLabelsTemp.col(i));
+
+  arma::mat output, XRbm(hiddenLayerSize, trainData.n_cols),
+      YRbm(hiddenLayerSize, testLabels.n_cols);
+
+  XRbm.zeros();
+  YRbm.zeros();
+
+  BinaryRBMPolicy<> binary_rbm(trainData.n_rows, hiddenLayerSize);
+  GaussianInitialization gaussian(0, 0.1);
+  RBM<GaussianInitialization, BinaryRBMPolicy<>> model(trainData,
+      gaussian, binary_rbm, 1,  1, true, false);
+
+  size_t numRBMIterations = trainData.n_cols * numEpoches;
+  numRBMIterations /= batchSize;
+
+  StandardSGD msgd(0.06, batchSize, numRBMIterations, 0, true);
+  model.Reset();
+  model.Policy().VisibleBias().ones();
+  model.Policy().HiddenBias().ones();
+  // test the reset function
+  model.Train(trainData, msgd);
+
+  for (size_t i = 0; i < trainData.n_cols; i++)
+  {
+    model.Policy().HiddenMean(std::move(trainData.col(i)),
+        std::move(output));
+    XRbm.col(i) = output;
+  }
+
+  for (size_t i = 0; i < testData.n_cols; i++)
+  {
+    model.Policy().HiddenMean(std::move(testData.col(i)),
+      std::move(output));
+    YRbm.col(i) = output;
+  }
+  const size_t numClasses = 10; // Number of classes.
+  const size_t numBasis = 5; // Parameter required for L-BFGS algorithm.
+  const size_t numIterations = 100; // Maximum number of iterations.
+
+  // Use an instantiated optimizer for the training.
+  L_BFGS optimizer(numBasis, numIterations);
+  SoftmaxRegression regressor(trainData, trainLabels,
+      numClasses, 0.001, false, optimizer);
+
+  double classificationAccuray = regressor.ComputeAccuracy(testData,
+   testLabels);
+  std::cout << "Softmax Accuracy = " << classificationAccuray << std::endl;
+
+  L_BFGS rbmOptimizer(numBasis, numIterations);
+  SoftmaxRegression rbmRegressor(XRbm, trainLabels, numClasses,
+        0.001, false, rbmOptimizer);
+  double rbmClassificationAccuracy = rbmRegressor.ComputeAccuracy(YRbm,
+      testLabels);
+  std::cout << "RBM Accuracy = " << rbmClassificationAccuracy << std::endl;
+  BOOST_REQUIRE_GE(rbmClassificationAccuracy, classificationAccuray);
```

Comment:
@lozhnikov I was referring to this test:
```
Softmax Accuracy = 76.2382
RBM Accuracy = 70.0612
/home/travis/build/mlpack/mlpack/src/mlpack/tests/rbm_network_test.cpp(118): fatal error in "ClassificationTest": critical check rbmClassificationAccuracy >= classificationAccuray failed [70.061213132999441 < 76.238174735670569]
ssRBM Accuracy = 80.801335559265439
```

Reply:
@lozhnikov This test seemed to be passing before the rebase, when `MiniBatchSGD` was being used. Any idea what can be done here?